### PR TITLE
fix: handle failure state when getting all known contacts

### PR DESCRIPTION
**Pull request by trOnk12**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- AR is breaking on newest kalium develop

### Causes (Optional)

- GetAllKnownUsersUseCase returns now a Result that may fail

### Solutions

- Add progress and failure state to the UI 

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-741/2/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt [hunk 1]
```diff
@@ -97,6 +94,25 @@ class NewConversationViewModel
         }
     }
 
+    private suspend fun tryGetAllContacts() {
+        innerSearchPeopleState = innerSearchPeopleState.copy(allKnownContacts = SearchResultState.InProgress)
+
```
Comment (MohammedMokresh) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt:99]: I think here you can remove the with context scope as the function is suspend and you call it inside view model scope 

and another suggestion

```suggestion
getAllContacts().let{
when(it){
is GetAllContactsResult.Failure -> {}
is GetAllContactsResult.Success -> {}
}
}
```

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt [hunk 2]
```diff
@@ -97,6 +94,25 @@ class NewConversationViewModel
         }
     }
 
+    private suspend fun tryGetAllContacts() {
```
Comment (MohammedMokresh) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt:97]: I think getAllContacts better name than tryGet all contacts
or just all contacts so it won't conflict with the use case name

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt (hunk 1 above)
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt:99]: I'll stick with val result = getAllContacts() as I was doing it in all other places already :P

**Review by MohammedMokresh — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-741/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3804341)(3804341)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3804341)(3804341)-dev-release.apk)

**Pull request merged**